src/include/migraphx/matcher.hpp

```diff
@@ -896,10 +897,20 @@ inline auto has_value(T x, std::size_t atol_mult = 10, std::size_t rtol_mult = 1
             else
             {
                 auto eps = std::numeric_limits<type>::epsilon();
-                if(std::all_of(v.begin(), v.end(), [&](auto val) {
-                       return std::fabs(val - static_cast<type>(x)) <
-                              eps * (atol_mult + rtol_mult * std::fabs(val));
-                   }))
+                if(atol_mult == 0 and rtol_mult == 0)
```

Comment:
This should check that `atol_mult + rtol_mult * std::fabs(x)` is equal to 0. This can be computed as a variable so it doesnt need to be recomputed everytime in the `all_of` loop.

Reply:
Done